pow/src/main/java/tech/pegasys/teku/pow/AbstractMonitorableEth1Provider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow;
+
+import tech.pegasys.teku.infrastructure.time.TimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.util.config.Constants;
+
+public abstract class AbstractMonitorableEth1Provider implements MonitorableEth1Provider {
+  protected enum Result {
+    success,
+    failed
+  }
+
+  private final TimeProvider timeProvider;
+  protected UInt64 lastCallTime = UInt64.ZERO;
+  protected UInt64 lastValidationTime = UInt64.ZERO;
+  protected Result lastCallResult = Result.failed;
+  protected Result lastValidationResult = Result.failed;
```

Comment:
I suspect this wind up being access from multiple threads  so probably need to be `volatile`.  Or possibly simpler to make the public methods `synchronous` and not need to worry about concurrent updates to the variables.

Reply:
my personal habit is to synchronize only when required and accept non-critical misbehaviors that recover automatically. 
I think we are in this case here, but yeah, let's go safe.

Reply:
Unfortunately things can get a lot more unpredictable than most people imagine when thread safety isn't right.  You might just see a stale value and things will work out or you might see a partially updated value and all kinds of weirdness can result.  Those UInt64 values are a `long` under the hood so are stored in 2 bytes - you can wind up seeing one of those 2 bytes with an updated value and one with the stale value leading to quite strange numbers.

Reply:
Right! Thanks for explanation